twa

```diff
@@ -391,8 +392,78 @@ function stage_4_scm_repos {
 # * Fetch the server's cert using `openssl s_client`, check it for known bad suites
 #   * Alternatively, just `nmap --script ssl-enum-ciphers -p 443 $domain`
 
+function stage_5_CAA {
+  verbose "Stage 5: Certification Authority Authorization"
+  verbose "Which Certificate Authorities (CAs) are allowed to issue certificates for the domain"
+    code=`dig +short caa $domain`
+    number_of_lines=`echo $code|wc -l`
+    if [ $number_of_lines -gt 1 ];then
+      INFO "multiple CAAs specified, everyone join the party"
+      #Cloudflare currently set the following set of certs as CAA
+      if [ echo $code|grep -c 'issue “comodoca.com”' -gt 0 ];then
```

Comment:
This is difficult to read, and I'm unclear what the testing value of it is -- what security knowledge do we gain by learning that this CAA record was probably generated by Cloudflare?

(That's not a hard "no", I genuinely don't know. Either way, these checks should be chained together using `&&` for readability).

Reply:
Cloudflare has a set of rules that need to be configured, and if they are done incorrectly can cause things such as unintended caching of pages you didn't want cached. It is useful to know if a site is behind a CDN.

Reply:
That makes sense, thanks for the explanation!

As for making into this something other than `INFO`: is there any other information we can glean to provide more certainty about whether the caching was, in fact, done incorrectly? I'm hesitant to emit `MEH`s or `FAIL`s for things that *might* be the case.

Reply:
re caching, the only way would be to update a page and check it's behaviour according to the specific rules set. But knowing that it is behind cloudflare (or another CDN) gives you a hint of where to look for problems. If I'm inspecting a site, I like to know as much about it, it's hosting, and it's distribution as possible. I'd also like to add in a detection for geolocation content(different content shown for different regions), but that can only be done via a few proxies.